addons/purchase_stock/models/purchase.py

```diff
@@ -78,6 +87,15 @@ def write(self, vals):
                         to_log[order_line] = (order_line.product_qty, pre_order_line_qty[order_line])
                 if to_log:
                     order._log_decrease_ordered_quantity(to_log)
+        # if date_planned value is set, use this to set move line expected dates
+        if vals.get('date_planned'):
+            new_date = fields.Datetime.to_datetime(vals['date_planned'])
+            # let's make sure it's not "no date"
+            if new_date:
+                for order in self:
```

Comment:
Instead of the double for loop, you can use the implicit mapped + filtered directly on the order_line field.
```python
order_lines_to_update = self.order_line.filtered(lambda l: l.propagate_date)
for order_line in order_lines_to_update:
    order_line._update_move_expected_date(new_date)
```
And the solution avoids to get too much indent

Reply:
Nice. Also, you don't need to loop on the `order in self` either -> `self.order_line.filtered(lambda l: l.propagate_date)` will implicitly mapped order_line for each order in self. Right ?

Reply:
Dang, sorry I missed grabbing that part.